homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -88,7 +88,7 @@ async def async_step_confirm(self, user_input=None):
             step_id="confirm", description_placeholders=placeholders
         )
 
-    async def async_step_ssdp(self, discovery_info):
+    async def async_step_ssdp(self, discovery_info=None):
```

Comment:
When will this step be called without `discovery_info`?

Reply:
I agree with that comment, however, zeroconf & ssdp call:

https://github.com/home-assistant/core/blob/b15caf31a97cb136078f1c89b36372492288b644/homeassistant/data_entry_flow.py#L107-L109

Which calls:

https://github.com/home-assistant/core/blob/b15caf31a97cb136078f1c89b36372492288b644/homeassistant/data_entry_flow.py#L182-L188

For that reason, the base method is now implemented as:

```python
    async def async_step_discovery(
        self, user_input: Optional[Dict[str, Any]] = None
    ) -> Dict[str, Any]:
        """Handle a flow initialized by discovery."""
```

Hence the line you've commented has to match the parent signature, defaulting to `None`.

Reply:
I'm happy to solve that, because I agree it ain't nice. However, I don't see how without breaking out `async_init` or even `_async_handle_step` ?

Reply:
Does mypy complain if we change the signature of `async_step_discovery`? That method should not have `user_input` as optional. And we should also name the parameter `discovery_info` I think.

Reply:
I'm going to try 👍

Reply:
Well, that worked! Honestly, didn't expect that.

I've adjusted the signature in the base, and adjusted the whole codebase to match that for consistency and linting proposes.

fb17d12